.github/workflows/ant.yml

```diff
@@ -60,10 +57,105 @@ jobs:
       with:
         name: cldr-apps
         path: tools/cldr-apps/cldr-apps.war
+  surveytest:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        lfs: true # important!
+    - name: Set up JDK 1.8
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+    - name: Download cldr.jar
+      uses: actions/download-artifact@v2
+      with:
+        name: cldr-tools
+        path: tools/java/dist/for-build
+    - name: Download cldr-apps.war
+      uses: actions/download-artifact@v2
+      with:
+        name: cldr-apps
+        path: tools/java/dist/for-build
+    # Cache for Tomcat tarball (load on cache hit, else store at end of job)
+    - name: Cache for Tomcat tarball
+      id: cache-tomcat
+      uses: actions/cache@v2
+      with:
+        path: tomcat-tarball
+        key: ${{ runner.os }}-tomcat-tarball
+    - name: Download Tomcat # only on cache miss
+      if: steps.cache-tomcat.outputs.cache-hit != 'true'
+      run: 'mkdir -p ./tomcat-tarball && cd ./tomcat-tarball && wget -O - "https://www.apache.org/dyn/mirrors/mirrors.cgi?action=download&filename=tomcat/tomcat-8/v8.5.57/bin/apache-tomcat-8.5.57.tar.gz" | tar xfpz - ; cd .. '
+    - name: Alias Tomcat directory
+      run: ln -svf tomcat-tarball/apache-tomcat-* tomcat
+    - name: Survey Tool Check
+      run: ant -noinput -DCLDR_TOOLS=$(pwd)/tools/java -DCATALINA_HOME=$(pwd)/tomcat -f tools/cldr-apps/build.xml check
+  test:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        lfs: true # important!
+    - name: Set up JDK 1.8
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+    - name: Download cldr.jar
+      uses: actions/download-artifact@v2
+      with:
+        name: cldr-tools
+        path: tools/java/dist/for-build
+    - name: Debug
+      run: find tools/java/dist/for-build -ls
+    - name: Build tools/cldr-unittest
+      run: ant -noinput -f tools/cldr-unittest/build.xml -DCLDR_DIR=$(pwd) tests
     # Now run tests
     - name: CLDR Unit Test
       run: ant -noinput unittest -f tools/cldr-unittest/build.xml -DCLDR_DIR=$(pwd)
     - name: CLDR Data Check
       run: ant -noinput datacheck -f tools/cldr-unittest/build.xml -DCLDR_DIR=$(pwd)
-    - name: Survey Tool Check
-      run: ant -noinput -DCLDR_TOOLS=$(pwd)/tools/java -DCATALINA_HOME=$(pwd)/tomcat -f tools/cldr-apps/build.xml check
+  deploy:
+    needs:
+      - build
+      - surveytest
+      - test
+    runs-on: ubuntu-latest
+    steps:
+    - name: Download cldr-apps.war
+      uses: actions/download-artifact@v2
+      with:
+        name: cldr-apps
+    - name: Deploy to cldr-ref (experimental)
+      # Only on push.
+      if: ${{ github.event_name == 'push' }}
```

Comment:
Note: this workflow is configured to execute on push events only on the master branch (which is what gets triggered when merging a PR).  Which makes sense, as it does for certain other things (ex: we have our API docs re-generate only on merges to master).

The comment "Only on push" kinda of obfuscates that fact, so maybe reword?

Reply:
@echeran the point is that PRs don't cause a deployment, only a merge. The workflow is triggered on pr OR push.

Reply:
@echeran updated the wording of the comment.